### ISSUE TITLE
Mock server seems to not release resources with reset()

### ISSUE THREAD

**Issue by pmaslany** [mock-server/mockserver#742]
**Describe the issue**
Seems like mock server doesn't release network resources when reset() is called.
Issue appears only on Windows host (test on Windows 10 Pro).
Works fine on linux (Ubuntu and CentOs)

**What you are trying to do**
Running some automated test, that for a set of dates mocks multiply endpoints, performs processing and reset a mock before next iteration.

**MockServer version**
5.9.0

**To Reproduce**
Run attached unit test.

1. How you are running MockServer (i.e maven plugin, docker, etc)
Junit test.
2. Code you used to create expectations
```java
package dk.eg.sd.loen.reimbursement.domain;

import io.restassured.RestAssured;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Test;
import org.mockserver.integration.ClientAndServer;
import org.mockserver.matchers.Times;
import org.mockserver.model.HttpRequest;
import org.mockserver.model.HttpResponse;

class MockServerTest {

    private ClientAndServer server = ClientAndServer.startClientAndServer(9000);

    void setupMock(int i) {

        String path = "/test/"+i;
        HttpRequest request = HttpRequest.request(path).withHeader("testHeader", "test");
        HttpResponse response = HttpResponse.response("{\"body\": \"test\"}").withHeader("Content-Type", "application/json").withStatusCode(200);
        server.when(request, Times.unlimited()).respond(response);
    }

    @AfterEach
    void killMock() {
        server.stop();
    }

    @Test
    void mockServerHangTest() {

        for (int a = 1; a < 100; a++) {

            for (int i = 1; i < 300; i++) {
                setupMock(i);
            }
            RestAssured.with().port(9000)
                    .header("testHeader", "test")
                    .get("/test/1")
                    .then().statusCode(200)
                    .extract().body().asString();

            System.out.println("Executed iteration: " + a);

            server.reset();
        }
    }
}
```
3. What error you saw
```
Executed iteration: 1
....
Executed iteration: 52
Executed iteration: 53
org.mockserver.client.SocketConnectionException: Address already in use: no further information: /127.0.0.1:9000
	at org.mockserver.client.NettyHttpClient.sendRequest(NettyHttpClient.java:109)
	at org.mockserver.client.MockServerClient.sendRequest(MockServerClient.java:183)
	at org.mockserver.client.MockServerClient.sendExpectation(MockServerClient.java:800)
	at org.mockserver.client.ForwardChainExpectation.respond(ForwardChainExpectation.java:47)
	at dk.eg.sd.loen.reimbursement.domain.MockServerTest.setupMock(MockServerTest.java:20)
	at dk.eg.sd.loen.reimbursement.domain.MockServerTest.mockServerHangTest(MockServerTest.java:34)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:532)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:171)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:167)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:114)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:59)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$4(NodeTestTask.java:108)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:98)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:74)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$4(NodeTestTask.java:112)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:98)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:74)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$4(NodeTestTask.java:112)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:98)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:74)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: io.netty.channel.AbstractChannel$AnnotatedSocketException: Address already in use: no further information: /127.0.0.1:9000
Caused by: java.net.BindException: Address already in use: no further information
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:327)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:336)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:685)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
**Expected behaviour**
Executed loop over 100 times.

**MockServer Log**

**Comment by jamesdbloom** [comment]
`reset` only clears expectations and logs and has no impact on ports, see https://mock-server.com/mock_server/clearing_and_resetting.html.

It looks to me that RestAssured is doing something with your ports so that no new connection requests can be received on port 9000 in your operation system.  This looks like a Windows combined with RestAssured problem and not related to MockServer, so I am going to close this issue.  If you have reason to demonstrate otherwise please open a new issue.

I can't reproduce this with the following code:
```java
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Test;
import org.mockserver.integration.ClientAndServer;
import org.mockserver.matchers.Times;
import org.mockserver.model.HttpRequest;
import org.mockserver.model.HttpResponse;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Map;

import static org.mockserver.serialization.ObjectMapperFactory.createObjectMapper;

class MockServerTest {

    private ClientAndServer server = ClientAndServer.startClientAndServer(9000);
    private ObjectMapper objectMapper = createObjectMapper();

    void setupMock(int i) {

        String path = "/test/" + i;
        HttpRequest request = HttpRequest.request(path);
        HttpResponse response = HttpResponse.response("{\"body\": \"test\"}").withHeader("Content-Type", "application/json").withStatusCode(200);
        server.when(request, Times.unlimited()).respond(response);
    }

    @AfterEach
    void killMock() {
        server.stop();
    }

    @Test
    void mockServerHangTest() throws IOException {

        for (int a = 1; a < 100; a++) {

            for (int i = 1; i < 300; i++) {
                setupMock(i);
            }
            HttpURLConnection connection = (HttpURLConnection) new URL("http://localhost:" + 9000 + "/test/1").openConnection();
            connection.setRequestMethod("GET");
            objectMapper.readValue(connection.getInputStream(), Map.class);

            System.out.println("Executed iteration: " + a);

            server.reset();
        }
    }
}
```

**Comment by pmaslany** [comment]
James, it's not an issue with RestAssured. The problem happens also when don't do any calls to mocked service. Finally I've found a solution. 
Seems like OS releases resources (sockets on port) too slow and we can reach a limit. Then web socket client can't connect to the server. That ends up with "Address already in use". As I stated it happens only on Windows.
In my case I applied some tuning for tcpip based on https://deploymentresearch.com/fix-for-windows-10-exhausted-pool-of-tcpip-ports/ and it solved an issue.
